GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -0,0 +1,89 @@
+﻿using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Common.Maintenance
+{
+    public class LooseObjectsStep : GitMaintenanceStep

```

Comment:
Hrmm maybe some comments would be most helpful here.  I've added some. 

The class does all maintenance associated with LooseObjects so I think the name is appropriate.

Reply:
`LooseObjectsCleanupStep`?

Reply:
I added the comments, there will be more comments as we expand what the Step does.

Part2 of this will take loose objects and put them in packfiles.

Let me know if anything seems confusing!

Reply:
I guess my question would be is this same class going to be doing the packing of loose objects as well or will that be a separate maintenance step?

Reply:
It needs to be the same step, because if they are scheduled independently we could have loose objects be packed (the next part of this step) and then deleted soon after (this part of the step). By doing both (i) deleting and _then_ (ii) packing, we can guarantee that we are not interrupting Git commands that don't see the new pack-file and try to read from a loose object on disk.

A similar ordering is required for the packfile cleanup. See #488 for more discussion on this.